src/components/Sources.js

```diff
@@ -15,12 +16,61 @@ const Outline = createFactory(_Outline);
 import _SourcesTree from "./SourcesTree";
 const SourcesTree = createFactory(_SourcesTree);
 
+type SourcesState = {
+  selectedPane: string
+};
+
 class Sources extends Component {
   renderShortcut: Function;
+  selectedPane: String;
+  togglePane: Function;
+  renderFooter: Function;
+  renderChildren: Function;
+  state: SourcesState;
 
   constructor(props) {
     super(props);
+    this.state = { selectedPane: "sources" };
+
     this.renderShortcut = this.renderShortcut.bind(this);
+    this.togglePane = this.togglePane.bind(this);
+    this.renderFooter = this.renderFooter.bind(this);
+  }
+
+  togglePane() {
+    const selectedPane = this.state.selectedPane === "sources"
+      ? "outline"
+      : "sources";
+
+    this.setState({ selectedPane });
+  }
+
+  renderFooter() {
+    const { selectedPane } = this.state;
+    const showSourcesTooltip = L10N.getStr("sourcesPane.showSourcesTooltip");
+    const showOutlineTooltip = L10N.getStr("sourcesPane.showOutlineTooltip");
+    const tooltip = selectedPane === "sources"
```

Comment:
we are evaluating if selectedPane === "sources" here and at 55 -- lets save this as a variable called "isSourcesPaneSelected". or something